docs/howtos/exposing-rust-components-to-kotlin.md

```diff
@@ -0,0 +1,203 @@
+
+## How to expose your Rust Component to Kotlin
+
+Welcome!
+
+It's great
+that you've built a Rust Component
+and want to expose it to Koltin,
+but we haven't written a complete guide yet.
+
+Please share your :+1:
+on [the relevant github issue](https://github.com/mozilla/application-services/issues/599)
+to let us know
+that you wanted it.
+
+In the meantime,
+here are some preliminary notes:
+
+### High-level overview.
+
+The [fxa-client](/components/fxa-client) component
+provides a useful example.
+We assume you have written
+a nice core of Rust code
+in a [./src/](/components/fxa-client/src) directory
+for your component.
+
+First, you will need to
+flatten the Rust API
+into a set of C FFI bindings,
+conventially in a [./ffi/](/components/fxa-client/ffi) directory.
+Use the [ffi_support](https://docs.rs/ffi-support/0.1.3/ffi_support/) crate
+to help make this easier,
+and consule that crate's documentation
+for tips and gotchas.
+
+Next, you will need to
+write Kotlin code
+that consumes that C FFI,
+conventionally in a [./android/](/components/fxa-client/android) directory.
+This code should use [JNA](https://github.com/java-native-access/jna)
+to load the C FFI via shared library
+and expose it
+as a nice safe ergonomic Kotlin API.
+
+It seems likely
+that we could provide
+a useful template
+and/or suite of shared utilities
+to make the Kotlin easier to write.
+But we haven't yet.
+
+Finally, you will need to
+add your package
+into the [android-components](https://github.com/mozilla-mobile/android-components) repo
+via some undocumented process
+that starts by asking
+in the rust-components channel on slack.
+
+### How should I name the resulting package?
```

Comment:
I can probably come up with answers to most of these questions fairly easily. Feel free to add any others you'd like answered, and I'll get something up after this lands. I was having trouble coming up with the questions that need answering when writing in the google doc some of this is based on.